pod/annotations_test.go

```diff
@@ -72,3 +74,122 @@ func TestPodPlatformSidecars(t *testing.T) {
+			},
+			wantSidecars: []PlatformSidecar{
+				{
+					Name:     "healthz-v2",
+					Channel:  "staging",
+					ArgsJSON: []byte(`{"cmd": "echo", "cmdArgs": "healthy"}`),
+				},
+			},
+		},
+		{
+			desc: "one platform sidecar set to false",
+			annotations: map[string]string{
+				"healthz-v1.platform-sidecars.netflix.com":         "false",
+				"healthz-v1.platform-sidecars.netflix.com/channel": "stable",
+			},
+			wantSidecars: nil,
+		},
+		{
+			desc: "many platform sidecars",
+			annotations: map[string]string{
+				"healthz-v1.platform-sidecars.netflix.com":           "false",
+				"healthz-v1.platform-sidecars.netflix.com/channel":   "stable",
+				"pod.netflix.com/cpu-bursting-enabled":               "true",
+				"healthz-v2.platform-sidecars.netflix.com":           "true",
+				"healthz-v2.platform-sidecars.netflix.com/channel":   "staging",
+				"healthz-v2.platform-sidecars.netflix.com/arguments": `{"cmd": "echo", "cmdArgs": "healthy"}`,
+				"healthz-v3.platform-sidecars.netflix.com":           "true",
+				"healthz-v3.platform-sidecars.netflix.com/channel":   "stable",
+			},
+			wantSidecars: []PlatformSidecar{
+				{
+					Name:     "healthz-v2",
+					Channel:  "staging",
+					ArgsJSON: []byte(`{"cmd": "echo", "cmdArgs": "healthy"}`),
+				},
+				{
+					Name:    "healthz-v3",
+					Channel: "stable",
+				},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.desc, func(t *testing.T) {
+			sidecars, err := PlatformSidecars(tt.annotations)
+			assert.NilError(t, err, "PlatformSidecars(%+v)", tt.annotations)
+
+			lessThan := func(a, b PlatformSidecar) bool {
+				return a.Name < b.Name
+			}
+			assert.Check(t, cmp.DeepEqual(tt.wantSidecars, sidecars, cmpopts.SortSlices(lessThan)), "PlatformSidecars(%+v)", tt.annotations)
+		})
+	}
+}
+
+func TestPlatformSidecarsErrors(t *testing.T) {
+	tests := []struct {
+		desc            string
+		annotations     map[string]string
+		wantErrContains string
+	}{
+		{
+			desc: "sidecar include annotation is not a bool",
+			annotations: map[string]string{
+				"healthz-v1.platform-sidecars.netflix.com": "maybe",
+			},
+			wantErrContains: `sidecar annotation "healthz-v1.platform-sidecars.netflix.com" must be a bool value`,
+		},
+		{
+			desc: "sidecar is missing channel annotation",
+			annotations: map[string]string{
+				"healthz-v1.platform-sidecars.netflix.com":           "true",
+				"healthz-v1.platform-sidecars.netflix.com/arguments": `{"cmd": "echo", "cmdArgs": "healthy"}`,
+			},
+			wantErrContains: `sidecar "healthz-v1.platform-sidecars.netflix.com" must have a channel specified via annotation "healthz-v1.platform-sidecars.netflix.com/channel"`,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.desc, func(t *testing.T) {
+			_, err := PlatformSidecars(tt.annotations)
+			assert.Check(t, cmp.ErrorContains(err, tt.wantErrContains), "PlatformSidecars(%+v)", tt.annotations)
```

Comment:
Can we use `testify/assert` instead of vendoring this new library, for consistency with the existing tests in this repo?

Reply:
I was already being consistent because the existing tests in this file were already using the `gotest.tools/assert` library. Other tests in this repo use `testify/assert`, so there are two assertions libraries being used in this repo.

`gotest.tools/assert` is actually more powerful because it uses `go-cmp` for [deep equals](https://pkg.go.dev/gotest.tools/assert/cmp#DeepEqual), which allows you to do things like ignore order when diffing slices and/or treat nil slices the same as empty slices. AFAIK `testify/assert` does not allow you to easily do those things.

Reply:
Okay, seems like the inconsistency is not your fault, then :).

Reply:
I'm ok with whatever.